src/group_theory/submonoid/pointwise.lean

```diff
@@ -300,3 +304,70 @@ end group_with_zero
 open_locale pointwise
 
 end add_submonoid
+
+/-! ### Elementwise multiplication of two additive submonoids
+
+These definitions are a cut-down versions of the ones around `submodule.has_mul`, as that API is
+usually more useful. -/
+namespace add_submonoid
+
+variables [non_unital_non_assoc_semiring R]
+
+/-- Multiplication of additive submonoids of a semiring R. The additive submonoid `S * T` is the
+smallest R-submodule of `R` containing the elements `s * t` for `s ∈ S` and `t ∈ T`. -/
+instance : has_mul (add_submonoid R) :=
+⟨λ M N, ⨆ s : M, N.map $ add_monoid_hom.mul s.1⟩
+
+theorem mul_mem_mul {M N : add_submonoid R} {m n : R} (hm : m ∈ M) (hn : n ∈ N) : m * n ∈ M * N :=
+(le_supr _ ⟨m, hm⟩ : _ ≤ M * N) ⟨n, hn, rfl⟩
+
+theorem mul_le {M N P : add_submonoid R} : M * N ≤ P ↔ ∀ (m ∈ M) (n ∈ N), m * n ∈ P :=
+⟨λ H m hm n hn, H $ mul_mem_mul hm hn,
+λ H, supr_le $ λ ⟨m, hm⟩, map_le_iff_le_comap.2 $ λ n hn, H m hm n hn⟩
+
+@[elab_as_eliminator] protected theorem mul_induction_on
+  {M N : add_submonoid R}
+  {C : R → Prop} {r : R} (hr : r ∈ M * N)
+  (hm : ∀ (m ∈ M) (n ∈ N), C (m * n))
+  (ha : ∀ x y, C x → C y → C (x + y)) : C r :=
+(@mul_le _ _ _ _ ⟨C, by simpa only [zero_mul] using hm _ (zero_mem _) _ (zero_mem _), ha⟩).2 hm hr
+
+open_locale pointwise
+
+variables R
```

Comment:
Regex on codebase says omitting brackets is very unidiomatic.
```suggestion
variables (R)
```
I note that this is copied from the line above `submodule.span_mul_span`. Maybe if you add brackets here, then add them there too.

Reply:
I agree brackets are more idiomatic, but I'm now on mobile and can't edit the other file easily. Let's do a bracket cleanup pass in a future PR, and leaves this consistent for now.